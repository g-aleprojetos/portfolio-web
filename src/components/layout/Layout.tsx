import React, {ReactNode} from 'react';
import {useBackgroundContext} from 'context/background';
import * as S from './Layout.styles';

type ILayout = {
  children: ReactNode;
};

export const Layout = ({children}: ILayout) => {
  const {themeDark} = useBackgroundContext();
  return (
    <S.Container data-testid={'layout-page'} data-backgroundblack={themeDark}>
      {children}
    </S.Container>
  );
};
