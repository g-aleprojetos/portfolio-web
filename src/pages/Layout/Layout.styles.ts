import styled from 'styled-components';
import {colors} from 'resources/colors';

export interface ExtraProps {
  $backgroundblack: boolean;
}

export const Container = styled.div<ExtraProps>`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  background-color: ${({$backgroundblack}) =>
    $backgroundblack ? colors.background02 : colors.darkgray};
`;

export const Content = styled.div`
  display: flex;
  position: fixed;
  width: 100%;
  height: 100%;
  margin-top: 116px;
  margin-bottom: 70px;
  overflow-y: auto;
`;
