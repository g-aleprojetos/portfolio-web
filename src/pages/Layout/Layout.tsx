import React from 'react';
import {Outlet} from 'react-router-dom';
import {useBackgroundContext} from 'context/background';
import {Header} from 'components/header';
import {Footer} from 'components/footer';
import 'utils/i18n';
import * as S from './Layout.styles';

export const Layout = () => {
  const {themeDark} = useBackgroundContext();

  return (
    <S.Container data-testid={'layout-page'} $backgroundblack={themeDark}>
      <Header />
      <S.Content>
        <Outlet />
      </S.Content>
      <Footer />
    </S.Container>
  );
};
