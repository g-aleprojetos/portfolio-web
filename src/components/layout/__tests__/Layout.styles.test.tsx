import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../Layout.styles';

describe('Layout.styles', () => {
  test('Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a true', () => {
    const {container} = render(<S.Container data-backgroundblack={true} />);
    expect(container).toMatchSnapshot();
  });

  test('Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a false', () => {
    const {container} = render(<S.Container data-backgroundblack={false} />);
    expect(container).toMatchSnapshot();
  });
});
