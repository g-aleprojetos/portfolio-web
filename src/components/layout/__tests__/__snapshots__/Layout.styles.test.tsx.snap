// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout.styles Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a false 1`] = `
.c0 {
  text-align: center;
  min-height: 100vh;
  background-color: #202024;
}

<div>
  <div
    class="c0"
    data-backgroundblack="false"
  />
</div>
`;

exports[`Layout.styles Container DEVE ser igual ao snapshot QUANDO data-backgroundblack for igual a true 1`] = `
.c0 {
  text-align: center;
  min-height: 100vh;
  background-color: #d3d3d3;
}

<div>
  <div
    class="c0"
    data-backgroundblack="true"
  />
</div>
`;
